src/care/LoopFuser.cpp

```diff
@@ -34,24 +34,23 @@ LoopFuser::LoopFuser() :
    m_lambda_reserved(0),
    m_lambda_size(0),
    m_lambda_data(nullptr),
-   m_is_scan(false),
-   m_is_counts_to_offsets_scan(false),
    m_scan_pos_outputs(nullptr),
    m_scan_pos_starts(nullptr),
    m_verbose(false),
    m_reverse_indices(false) {
+
+      // Supports fusing up to 10k loops of average lambda size of 256 bytes
+      // will flush if we exceed the 10k count or if the lambda size requirements
+      // are exceeded.
+      reserve(10*1024);
+      reserve_lambda_buffer(256*10*1024);
 }
 
 LoopFuser * LoopFuser::getInstance() {
    static LoopFuser * instance = nullptr;
    if (instance == nullptr) {
       instance = new LoopFuser();
-
-      // Supports fusing up to 1M loops of average lambda size of 256 bytes
-      // will flush if we exceed the 1M count or if the lambda size requirements
-      // are exceeded.
-      instance->reserve(1024*1024);
-      instance->reserve_lambda_buffer(256*1024*1024);
+      defaultObserver->registerFusedActions(instance, -FLT_MAX/2);
```

Comment:
why is this value different from the default value?

Reply:
this is to avoid the default actions triggering the warning of registering a priority after / the same as the previous priority.

Reply:
A comment here would be useful.

Is the intent here to always be greater than the existing one?  If we call getInstance(), delete it, and call getInstance() again we will get the warning, right?

Reply:
yes.